### ISSUE TITLE
Add option for deterministic tie-breaking in tf.math.reduce_min

### ISSUE THREAD

**Issue by dan-kazbek** [tensorflow/tensorflow#97688]
### Issue type

Feature Request

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

tf 2.19

### Custom code

Yes

### Python version

3.12

### Current behavior?

Currently, the behavior of `tf.math.reduce_min` is that when there are multiple equal minimum elements, the backward pass splits the upstream gradient equally among them. This behavior is neither documented nor immediately obvious. It would be helpful to have an option for controlling how gradients are handled in case of ties—either splitting them evenly or redirecting the entire gradient to the first minimum element (similar to how `tf.argmin` works). At the very least, the current behavior should be documented.

Attached is a standalone example demonstrating how the gradients for a function using `tf.reduce_min` differ from those for a function using `tf.argmin` to achieve the same computation.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

def compute_using_reduce_min(tensor):
    minimums = tf.reduce_min(
        input_tensor=tensor, axis=-2
    )
    return tf.reduce_sum(minimums, axis=-1)

def compute_using_argmin(tensor):
    indices = tf.argmin(tensor, axis=-2)
    col_indices = tf.range(tf.shape(tensor)[1], dtype=indices.dtype)
    full_indices = tf.stack([indices, col_indices], axis=1)
    minimums = tf.gather_nd(tensor, full_indices)
    return tf.reduce_sum(minimums, axis=-1)

def main():
    
    tensor = tf.Variable(tf.constant([[1, 2, 6], [4, 2, 3]], dtype=tf.float32))

    with tf.GradientTape(persistent=True) as tape:
        argmin_out = compute_using_argmin(tensor)
        reduce_min_out = compute_using_reduce_min(tensor)

    argmin_grads = tape.gradient(argmin_out, tensor)
    print("argmin grads: ", argmin_grads)
    reduce_min_grads = tape.gradient(reduce_min_out, tensor)
    print("reduce_min grads: ", reduce_min_grads)

    print("outputs are equal: ", tf.reduce_all(tf.math.equal(argmin_out, reduce_min_out)).numpy())
    print("gradients are equal: ", tf.reduce_all(tf.math.equal(argmin_grads, reduce_min_grads)).numpy())


if __name__ == "__main__":
    main()
```

### Relevant log output

```shell
argmin grads:  tf.Tensor(
[[1. 1. 0.]
 [0. 0. 1.]], shape=(2, 3), dtype=float32)
reduce_min grads:  tf.Tensor(
[[1.  0.5 0. ]
 [0.  0.5 1. ]], shape=(2, 3), dtype=float32)
outputs are equal:  True
gradients are equal:  False
```

**Comment by Venkat6871** [comment]
I was able to reproduce the same issue using TensorFlow 2.19.0 version. Please find the [gist](https://colab.sandbox.google.com/gist/Venkat6871/b43b5ff0803f632a3faecbd4c3adbc8b/97688_tf_2-19-0-v.ipynb) for your reference.
Thank you!
